src/hover_provider/modelHoverProvider.ts

```diff
@@ -0,0 +1,158 @@
+            document.uri.fsPath,
+        );
+        return;
+      }
+      if (word !== undefined && hover !== "ref") {
+        const dbtModel = word.match(ModelHoverProvider.GET_DBT_MODEL);
+        if (dbtModel && dbtModel.length === 1) {
+          const mdString = this.getDefinitionFor(
+            project.getProjectName(),
+            dbtModel[0],
+            document.uri,
+          );
+          if (mdString !== undefined) {
+            const hover = new Hover(mdString, new Range(position, position));
+            resolve(hover);
+          }
+          this.telemetry.sendTelemetryEvent("provideModelDefinition", {
+            type: "single",
+          });
+          return;
+        }
+        if (dbtModel && dbtModel.length === 3) {
+          const mdString = this.getDefinitionFor(
+            dbtModel[0],
+            dbtModel[2],
+            document.uri,
+          );
+          if (mdString !== undefined) {
+            const hover = new Hover(mdString, new Range(position, position));
+            resolve(hover);
+          }
+          this.telemetry.sendTelemetryEvent("provideModelDefinition", {
+            type: "dual",
+          });
+          return;
+        }
+      }
+      resolve(undefined);
+    });
+  }
+
+  private onManifestCacheChanged(event: ManifestCacheChangedEvent): void {
+    event.added?.forEach((added) => {
+      this.modelToLocationMap.set(added.projectRoot.fsPath, added.nodeMetaMap);
+    });
+    event.removed?.forEach((removed) => {
+      this.modelToLocationMap.delete(removed.projectRoot.fsPath);
+    });
+  }
+
+  private getDefinitionFor(
+    projectName: string,
+    modelName: string,
+    currentFilePath: Uri,
+  ): MarkdownString | undefined {
+    const projectRootpath =
+      this.dbtProjectContainer.getProjectRootpath(currentFilePath);
+    if (projectRootpath === undefined) {
+      return;
+    }
+    const nodeMap = this.modelToLocationMap.get(projectRootpath.fsPath);
+    if (nodeMap === undefined) {
+      return;
+    }
+    const node = nodeMap.get(modelName);
+    if (node) {
+      const content = new MarkdownString();
+      content.supportHtml = true;
+      content.isTrusted = true;
+      content.appendMarkdown(
+        `<span style="color:#347890;">(ref)&nbsp;</span><span><strong>${node.alias}</strong></span>`,
+      );
+      if (node.description !== null) {
+        content.appendMarkdown(`</br><span>${node.description}</span>`);
+      }
+      content.appendText("\n");
+      content.appendText("\n");
+      content.appendMarkdown("---");
+      content.appendText("\n");
+      content.appendText("\n");
+      for (const colKey in node.columns) {
+        const column = node.columns[colKey];
+        content.appendMarkdown(
+          `<br/><span style="color:#347890;">(column)&nbsp;</span><span>${column.name} &nbsp;</span>`,
+        );
+        if (column.data_type !== null) {
```

Comment:
Could you make this view more compact? What are the limits? How many lines can be displayed. Is there any way to expand it?

Reply:
I'll have to look into this. It does scrolling by default. The first screen shot in the description has scrolling enabled.

Reply:
Uploading Screen Recording 2023-09-17 at 11.20.41 AM.mov…

Reply:
https://github.com/AltimateAI/vscode-dbt-power-user/assets/79863718/e9e7abd4-8277-4abc-8e68-721bb4c543d6

Reply:
It actually fits the screen width:
https://github.com/AltimateAI/vscode-dbt-power-user/assets/79863718/15702629-c9a9-46ab-a656-a0376a1ccfcf